webpack/webpack-embedded-dependencies-plugin/src/EmbeddedDependenciesWebpackPlugin.ts

```diff
@@ -0,0 +1,383 @@
+ * const licenseMarkdownFileName: LicenseFileName = 'custom-license-file-name.md';
+ * const licenseTextFileName: LicenseFileName = 'custom-license-file-name.txt';
+ * ```
+ */
+export type LicenseFileName = `${string}.${'html' | 'md' | 'txt'}`;
+
+type PackageNameAndVersion = `${string}@${string}`;
+type ThirdPartyPackageMap = Map<
+  PackageNameAndVersion,
+  { packageFolderPath: string; packageJsonData: IPackageData }
+>;
+type DefaultLicenseTemplate = `<hr />${string}<br /><br />${string}`;
+
+/**
+ * @beta
+ * Webpack plugin that generates a file with the list of embedded dependencies
+ * and their licenses.
+ */
+export default class EmbeddedDependenciesWebpackPlugin implements WebpackPluginInstance {
+  private readonly _outputFileName: string;
+  private readonly _generateLicenseFile: boolean;
+  private readonly _generateLicenseFileFunction: LicenseFileGeneratorFunction;
+  private readonly _generatedLicenseFilename: LicenseFileName;
+  private readonly _packageFilterFunction: (packageJson: IPackageData, filePath: string) => boolean;
+
+  public constructor(options?: IEmbeddedDependenciesWebpackPluginOptions) {
+    this._outputFileName = options?.outputFileName || DEFAULT_EMBEDDED_DEPENDENCIES_FILE_NAME;
+    this._generateLicenseFile = options?.generateLicenseFile || false;
+    this._generateLicenseFileFunction =
+      options?.generateLicenseFileFunction || this._defaultLicenseFileGenerator;
+    this._generatedLicenseFilename = options?.generatedLicenseFilename || DEFAULT_GENERATED_LICENSE_FILE_NAME;
+    this._packageFilterFunction = options?.packageFilterPredicate || DEFAULT_PACKAGE_FILTER_FUNCTION;
+  }
+
+  /**
+   * @beta
+   * Webpack plugin apply method. This method is called by the webpack compiler to apply the plugin, however it not usually
+   * needed to be invoked manually by the developer in a webpack configuration. However, if you are calling this plugin (applying it from another plugin)
+   * you can call `plugin.apply(compiler)` to apply the plugin and invoke it.
+   * @param compiler - The webpack compiler instance.
+   */
+  public apply(compiler: Compiler): void {
+    // Tap into compilation so we can tap into compilation.hooks.processAssets
+    compiler.hooks.thisCompilation.tap(PLUGIN_NAME, (compilation, { normalModuleFactory }) => {
+      const thirdPartyPackages: ThirdPartyPackageMap = new Map();
+
+      normalModuleFactory.hooks.module.tap(
+        PLUGIN_NAME,
+        (module, moduleCreateData: IWebpackModuleCreateData, resolveData) => {
+          const { resourceResolveData } = moduleCreateData;
+          const pkg: IPackageData | undefined = resourceResolveData?.descriptionFileData;
+          const filePath: string | undefined = resourceResolveData?.descriptionFileRoot;
+
+          if (
+            pkg &&
+            filePath &&
+            this._packageFilterFunction(pkg, filePath) &&
+            filePath?.includes('node_modules')
```

Comment:
If this situation arises, we could implement an API which manually declares packages like this into the configuration (like externals etc.). What I don't want for the user: 

* Extra overhead of providing an more strict filter function because the default filter is overly loose. 
* Overly strict filter which doesn't provide flexibility

I think this strikes the balance for now, however if a need for this scenario arises we can add it.